src/main/java/frc/robot/commands/ResetEncoderAtBottom.java

```diff
@@ -0,0 +1,33 @@
+package frc.robot.commands;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.constants.ElevatorConstants;
+import frc.robot.subsystems.Elevator;
+
+public class ResetEncoderAtBottom extends CommandBase {
```

Comment:
I think it would make sense to combine this and the ElevatorCalibrateAbsoluteEncoder into just CalibrateElevatorZero. Is there a scenario where you would want to calibrate the motor encoder but not the absolute encoder? The absolute encoder is mainly just to store the position of the elevator even when the robot is off.

Reply:
I think knowledge of the limit switches should be in the subsystem. Say a zero method that gets invoked by a command. The subsystem should continually monitor the limit switches and do something appropriate when the a limit switch is tripped.

The command sets the motor power at the start, but what happens if the subsystem is running PID? (The tradition elevator subsystem runs PID all the time.) During the next call to periodic(), the subsystem would set a different power.

You have put the PID loop into the MoveToHeight() command. That confuses subsystems and commands. Say you are a subsystem. If I give you a command to move left 3 feet. You will do the moving. I should not be instructing something like a motor to push you 3 feet to the left.

Furthermore, think about how commands are used. There might be the sequential commands

MoveToHeight(0.85);
DropTheCone().

We need the MoveToHeight() to terminate when it gets to 0.85. But MoveToHeght never terminates, so we never drop the cone.

If MoveToHeight() terminated at 0.85, then it would stop and the cone would drop. But it would also stop running the PID loop, so the elevator will move from 0.85. Consider the sequence

MoveToHeight(0.85)
DoSomethingImportantForTheNext5Seconds()
DropTheCone()

We may drop the cone in the wrong place because the elevator moved.

Reply:
Sounds good, I will keep knowledge of limit switches within the subsystem.